### PR TITLE
Added Batch Call Tests [APPAI-1464]

### PR DESCRIPTION
**Pull request by sawood14012**
# Description
 Implementation of Batch call e2e tests for component analysis 
 https://issues.redhat.com/browse/APPAI-1464

# Feature: Component analysis V2 Batch Call API
_Scenarios covered_
- Check the component analysis V2 REST API Batch Call for NPM ecosystem
- Check the component analysis V2 REST API Batch Call for Maven ecosystem
- Check the component analysis V2 REST API Batch Call for Pypi ecosystem
- Check the component analysis V2 REST API Batch Call for unknown ecosystem
- Check the component analysis V2 REST API Batch Call for unknown packages
- Check the component analysis V2 REST API Batch Call for 1 known 1 unknown package

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Review by deepak1725 — CHANGES_REQUESTED** [review]
Check for CA no Vulnerability scenario,
Invalid request,

File: integration-tests/features/component_analysis_v2_batchcall.feature [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        Then I should get 200 status code
+         And I should receive a valid JSON response
+        Then I should be able to validate all the feilds or vulnerablities in the result
+        And I should find package sequence 3.0.0 has no recommendation
+        And I should find package ejs 1.0.0 has 3.1.3 recommended version
+        Then I should find snyk id SNYK-JS-EJS-10218 and 8.1 for package ejs and version 1.0.0
+        Then I should find snyk id SNYK-JS-ANGULAR-471882 for package angular and version 1.0.0 as private
+    
+     @cav2
+    Scenario: Check the component analysis V2 REST API Batch Call for Maven ecosystem
```
Comment (prashbnair) [inline integration-tests/features/component_analysis_v2_batchcall.feature:19]: Can you add a test for free tier and registered user

**Review by prashbnair — CHANGES_REQUESTED** [review]

File: integration-tests/features/component_analysis_v2_batchcall.feature (hunk 1 above)
Comment (sawood14012) [reply integration-tests/features/component_analysis_v2_batchcall.feature:20]: > Check for CA no Vulnerability scenario,
> Invalid request,

@deepak1725  Done added No Vulnerability scenario and invalid requests

Comment (sawood14012) [reply integration-tests/features/component_analysis_v2_batchcall.feature:20]: @prashbnair need to add a uuid in vault and modify the cent os ci jobs to check for registered user , for now i am checking only for free tier, will add registered user test cases after adding the User management tests

**Comment by sawood14012** [comment]
[test]

**Review by deepak1725 — APPROVED** [review]
LGTM, 
This PR https://github.com/fabric8-analytics/fabric8-analytics-server/pull/672 add some specs changes. Please ensure things are in Sync.

**Comment by sawood14012** [comment]
> LGTM,
> This PR [fabric8-analytics/fabric8-analytics-server#672](https://github.com/fabric8-analytics/fabric8-analytics-server/pull/672) add some specs changes. Please ensure things are in Sync.

okay will update accordingly

**Comment by sawood14012** [comment]
[test]

**Comment by sawood14012** [comment]
[test]

**Review by prashbnair — CHANGES_REQUESTED** [review]

File: integration-tests/features/steps/component_analysis_v2_batchcall.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    perform_CA_batch_call(context, use_user_key, file)
+
+
+@then('I should be able to validate all the feilds or vulnerablities in the result')
+def find_vulnerablities(context):
+    """I should be able to find some Vulnerablities."""
+    json_data = context.response.json()
+    for single_item in json_data:
+        if "vulnerability" in single_item:
+            assert "package" in single_item, "No package Found!"
```
[the remark this answers is not in the document]
Comment (deepak1725) [reply integration-tests/features/steps/component_analysis_v2_batchcall.py:104]: @prashbnair It means, If `package` is not in `single_line` then, Print this message in Assert Error Exception.

[the remark this answers is not in the document]
Comment (sawood14012) [reply integration-tests/features/steps/component_analysis_v2_batchcall.py:104]: yes if the key package is not in result then it will raise an exception

**Review by prashbnair — APPROVED** [review]

**Pull request merged**